exec/java-exec/src/main/java/org/apache/drill/exec/ExecConstants.java

```diff
@@ -146,89 +158,121 @@
    *                |-    bar  -  a.parquet
    *                |-    baz  -  b.parquet
    */
-  String FILESYSTEM_PARTITION_COLUMN_LABEL = "drill.exec.storage.file.partition.column.label";
-  OptionValidator FILESYSTEM_PARTITION_COLUMN_LABEL_VALIDATOR = new StringValidator(FILESYSTEM_PARTITION_COLUMN_LABEL, "dir");
+  public static final String FILESYSTEM_PARTITION_COLUMN_LABEL = "drill.exec.storage.file.partition.column.label";
+  public static final OptionValidator FILESYSTEM_PARTITION_COLUMN_LABEL_VALIDATOR = new StringValidator(
+      FILESYSTEM_PARTITION_COLUMN_LABEL, "dir",
+      "The column label for directory levels in results of queries of files in a directory. Accepts a string input.");
```

Comment:
Instead of saying that this takes a string, can we generate that part? We look up the default value, infer the type and say:

The column label ... Type: string, Default: "dir"

Or, add type and default as additional columns. (Or maybe the default does not matter, only the current value, so just list the type.)